core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataAsyncClient.java

```diff
@@ -116,11 +142,14 @@ private void get(String path, RetryPolicyContext retryPolicyContext, Completable
             log.debug(() -> "Retrying request: Attempt " + newAttempt);
             RetryPolicyContext newContext =
                 RetryPolicyContext.builder()
-                                  .request(baseTokenRequest)
                                   .retriesAttempted(newAttempt)
                                   .exception(SdkClientException.create(error.getMessage(), error))
                                   .build();
-            scheduledRetryAttempt(() -> get(path, newContext, returnFuture), newContext);
+            scheduledRetryAttempt(() -> {
+                // force token refresh on retryable error
+                this.tokenCache = createCachedTokenSupplier(this.tokenCacheStrategy);
```

Comment:
Just to double check, is this behavior documented in the specification? 
Hmm, tokenCache may need to be volatile.

Reply:
Yeah, I'm not to fond of this solution either... I will try to come up with something better.
The SEP mentions `A retry attempt MUST perform the entire get() call again, which means it MUST fetch a new session token if the current session token has expired` So I guess if the token is **NOT** expired, we don't have to fetch a new one? But should we still? It's not 100% clear to me...